core/chains/evm/forwarders/forwarder_manager.go

```diff
@@ -0,0 +1,308 @@
+package forwarders
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmlogpoller "github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/authorized_forwarder"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/authorized_receiver"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/offchain_aggregator_wrapper"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
+)
+
+// Config encompasses config used by fwdmgr
+//go:generate mockery --recursive --name Config --output ./mocks/ --case=underscore --structname Config --filename config.go
+type Config interface {
+	LogSQL() bool
+}
+
+var authTopics = []common.Hash{
```

Comment:
I don't think any of these need to be globals ? Also probably don't even need this authTopics var, I'd just do literal definition of []common.Hash where you need it for readability

Reply:
I moved all to where they are used except forwardABI as it's parsing JSON, it's better to be parsed once than with every tx imho